packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_analysis_model.dart

```diff
@@ -66,19 +58,30 @@ class FrameAnalysis {
 
   FramePhase _generateLayoutPhase() {
     final uiEvent = frame.timelineEventData.uiEvent;
-    if (uiEvent == null) {
-      return FramePhase.layout(events: <SyncTimelineEvent>[]);
+    if (uiEvent != null) {
+      final layoutEvent = uiEvent.firstChildWithCondition(
+        (event) =>
+            event.name
+                ?.caseInsensitiveEquals(FramePhaseType.layout.eventName) ??
+            false,
+      );
+
+      if (layoutEvent != null) {
+        final _buildChildren = layoutEvent.shallowNodesWithCondition(
+          (event) => event.name == FramePhaseType.build.eventName,
+        );
+        final buildDuration = _buildChildren.fold<Duration>(Duration.zero,
+            (previous, TimelineEvent event) {
+          return previous + event.time.duration;
+        });
+
+        return FramePhase.layout(
+          events: <SyncTimelineEvent>[layoutEvent as SyncTimelineEvent],
+          duration: layoutEvent.time.duration - buildDuration,
+        );
+      }
     }
-    final layoutEvent = uiEvent.firstChildWithCondition(
-      (event) =>
-          event.name?.caseInsensitiveEquals(FramePhaseType.layout.eventName) ??
-          false,
-    );
-    return FramePhase.layout(
-      events: <SyncTimelineEvent>[
-        if (layoutEvent != null) layoutEvent as SyncTimelineEvent,
-      ],
-    );
+    return FramePhase.layout(events: <SyncTimelineEvent>[]);
   }
```

Comment:
do not review this part here - this should be reviewed in https://github.com/flutter/devtools/pull/4327, and then will merge here once that PR lands.